src/modules/orders/server/utils.ts

```
@@ -537,3 +539,126 @@ export function mapOrderToBuyer(doc: Order): OrderForBuyer {
+
+/** If the user typed something that looks like an order code, return a normalized code. */
+function normalizeOrderCode(raw: string): string | null {
+  const withoutHash = raw.replace(/^#/, '');
+  // Adjust pattern to your format; this is permissive but safe
+  return /^[A-Za-z0-9\-_.]{4,40}$/.test(withoutHash) ? withoutHash : null;
+}
+
+export function buildSellerOrdersWhere(input: {
+  tenantId: string;
+  status?: Array<OrderStatus>;
+  query?: string;
```

Biome (2.1.2)

[error] Unexpected control character in a regular expression.

Control characters are unusual and potentially incorrect inputs, so they are disallowed.

(lint/suspicious/noControlCharactersInRegex)